### PR TITLE
fix(docs-infra): do not auto-link `http://` to the `common/http`

### LINKED ISSUE #31012: http text linking to http API
# 📚 Docs or angular.io bug report

### Description

"http" text for example purposes in the Location Strategy API page (http://localhost:4200/api/common/LocationStrategy#description) description is linking to the HTTP API (http://localhost:4200/api/common/http)

### What's the affected URL?**
http://localhost:4200/api/common/LocationStrategy#description

### Reproduction Steps**
- Go to http://localhost:4200/api/common/LocationStrategy#description
- Click on 'http' in http://example.com#/foo or http://example.com in description paragraph

### Expected vs Actual Behavior**
I would expect it to not go to the http API as it is just 'example text' and not pertaining to the http API

## 📷Screenshot
Screen Shot 2019-06-12 at 12 47 47 PM

### PR DESCRIPTION
**Pull request by gkalpak**
Previously, our auto-linking feature would match `http` in URLs (such as `http://...`) to the `common/http` package and automatically create a link to that, which was undesirable. While it is possible to work around that via `<code class="no-auto-link">http://...</code>`, most people didn't even realize the issue.

Since in this case it is possible to reliably know it is a false match, this commit fixes it by applying a custom auto-link filter that ignores all docs for `http`, if it comes before `://`.

Fixes #31012.

### CODE REVIEW

**Comment by gkalpak** [comment]
Example of the fix: The URLs in https://pr31051-37d4bba.ngbuilds.io/api/common/LocationStrategy#description.

File: aio/tools/transforms/angular-base-package/services/ignoreHttpInUrls.spec.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+const ignoreHttpInUrls = require('./ignoreHttpInUrls')();
+
+describe('ignoreHttpInUrls', () => {
+  it('should ignore all docs when matching `http` in `http://...`', () => {
+    const docs = [{ docType: 'package', name: 'http' }, { docType: 'class', name: 'Foo' }];
+    const filteredDocs = [];
+
+    const words1 = ['http', '://', 'example', '.', 'com', '/'];
+    expect(ignoreHttpInUrls(docs, words1, 0)).toEqual(filteredDocs);
```
Comment (petebacondarwin) [inline aio/tools/transforms/angular-base-package/services/ignoreHttpInUrls.spec.js]: NIT - I find the DRYness of using a variable `filteredDocs` in these tests make the tests less readable. I don't think there is any benefit in not repeating yourself here and below.

File: aio/tools/transforms/angular-base-package/services/ignoreHttpInUrls.spec.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    const docs = [{ docType: 'package', name: 'http' }, { docType: 'class', name: 'Foo' }];
+    const filteredDocs = [...docs];
+
+    const words1 = ['http', '://', 'example', '.', 'com', '/'];
+    expect(ignoreHttpInUrls(docs, words1, 2)).toEqual(filteredDocs);
+
+    const words2 = ['URL', ': ', 'http', '://', 'example', '.', 'com', '/'];
+    expect(ignoreHttpInUrls(docs, words2, 0)).toEqual(filteredDocs);
+
+    const words3 = ['https', '://', 'example', '.', 'com', '/'];
```
Comment (petebacondarwin) [inline aio/tools/transforms/angular-base-package/services/ignoreHttpInUrls.spec.js]: Perhaps we should also handle `https` for good measure? These are more likely to be used as URLs and you never know we might add a code item called `https`?

**Review by petebacondarwin — APPROVED** [review]
A couple of suggestions but nice work, thanks!

**Review by IgorMinar — APPROVED** [review]

File: aio/tools/transforms/angular-base-package/services/auto-link-filters/ignoreHttpInUrls.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+
+/**
+ * This service is used by the autoLinkCode post-processor to ignore the `http(s)` part in URLs
+ * (i.e. `http://...` or `https://...`).
+ */
+module.exports = function ignoreHttpInUrls() {
+  const ignoredSchemes = ['http', 'https'];
```
Comment (petebacondarwin) [inline aio/tools/transforms/angular-base-package/services/auto-link-filters/ignoreHttpInUrls.js]: FYI, I have generally taken the approach of adding such configurable values as properties on the service itself, which allows a config block to inject it and modify it, if desired.

**Pull request closed without merging**